### PR TITLE
Introduce energy requirements module to the package

### PR DESCRIPTION
**Pull request by elaouniyassine**
- Added core energy calculation functions for maintenance, activity, growth, lactation, work, fibre, pregnancy, total energy, meat, and DMI.

- Created internal orchestrator `run_energy_requirements()` that applies all energy computations rowwise on input data.

- Introduced robust input validations for all functions to enforce biological constraints.

- Added comprehensive tests covering exact formulas, species/cohorts, edge cases, and validation errors.

### CODE REVIEW

**Review by ahmedjoubest — CHANGES_REQUESTED** [review]
@elaouniyassine 

- [x] The nemeat function is now moved to the allocation. This should be adapted.
https://github.com/APPLITICS/gleam/pull/19/files
- [x] Reporting message (as discussed)

**Comment by ahmedjoubest** [comment]
@elaouniyassine 

- As Dominik suggested: add a placeholder function for negg and submit the PR again. Lydia will fill it with calculation once 
- Resolve conflicts and submit the PR again to Lydia for review.

image

**Comment by elaouniyassine** [comment]
Hi @lydialanzoni,

The energy requirements module is now ready for review.

## Key Improvements
### Core functionality (similar to herd module implementation)

- Introduced all energy-related core functions (maintenance, activity, growth, lactation, work, fibre, pregnancy, REM/REG, total energy and DMI).
- Each function includes input validation with `cli::cli_abort()` and argument checking.
- A central internal wrapper `run_energy_requirements()` orchestrates the full computation chain row-wise.

### Package structure

- Core logic is in `energy_requirements_core_model.R`
- The `run_energy_requirements()` internal function is located in `run_energy_requirements.R` and mirrors the structure of the herd simulation setup.

### Unit Testing

Extensive testthat suite added:
- Verifies formulas for each species/cohort using explicit expected values.
- Includes edge case and validation failure checks.
- Tests are deterministic and aligned with the scientific model.

### Documentation

- All core and wrapper functions are documented with `roxygen2`.
- Developers can update docs with `devtools::document()` if they edited something.

#### Validation considerations

Since this module is computation-heavy, validation is key. We'd like the team to test what we currently have and help decide if there are additional range checks or stricter input requirements we should enforce.

For example:
Should `diet_dig` always be checked to be between 0 and 1?
Should we explicitly stop on any NA values in key inputs, or allow them to propagate?

## Reminder: Pending chicken content
Chicken calculations aren’t included in the energy requirements module yet. As agreed with the team, these will be added later. A placeholder has been added in the meantime.

## Argument order and impact on accuracy
all functions were being called without naming the arguments, and several of them — specifically `Dfunction_nemain`, `Dfunction_nelact` and `Dfunction_nepreg` have a different order in the calls compared to how they are defined. 
Because R uses positional matching by default, this means arguments get mapped purely by position, not by name. If the call order doesn’t match the function definition, it silently produces incorrect results.

For example:
`Dfunction_getot(animal, cohort, nelact, nemain, ...)`
If `nelact`and `nemain`are swapped compared to the function signature, the calculation still runs, but it puts lactation energy where maintenance energy should be, and vice versa. This corrupts the entire energy requirement calculation without raising any errors.

As a result, the outputs are no longer identical to previous runs, and we cannot use the existing CSV (`GLEAM_input_directemissions.csv`) or rerun the old scripts as a benchmark to verify correctness.

For this reason, the improvements here are intentionally light, to stay as close as possible to the intended previous behavior, given that there was no valid reference output to compare against.

### Recommendation:
We strongly advise the team to review these implementations, run fresh recalculations, and test the outputs to ensure they still align with the expected model behavior.

File: legacy/Functions/03_functions_energyrequirements.R [hunk 1]
```diff
@@ -102,45 +96,40 @@ Dfunction_nemain = function(Animal_short, # CTL, BFL, SHP, GTS, PGS, CHK, CML
...
-                           nemain, # CTL, BFL, PGS, CHK
-                           average_weight, # SHP, GTS, CML
-                           offtake_rate # CTL, BFL, SHP, GTS, PGS, CHK, CML / offtake rate by cohort
-                           # activity_fraction,
-                           # high_activity_fraction
-
-){
-
-  if (Animal_short  %in% c("CTL", "BFL")){
+Dfunction_neact <- function(
```
Comment (lydialanzoni) [inline legacy/Functions/03_functions_energyrequirements.R:99]: @elaouniyassine 
Hi Yassine, I would like to introduce a small change here also to simplify a bit and streamline with the future user interface.
I would like to replace the mmspasture*past_man_frac with a variable called "activity_fraction" and  mmspasture * (1 - past_man_frac) with a variable called "high_activity_fraction".
This would mean that I would introduce this step in the PRE-PROCESSING, so that the input table would have those two variables as inputs (and no more past_man_frac). Let me know if I can proceed :)

Comment (lydialanzoni) [reply legacy/Functions/03_functions_energyrequirements.R:99]: Could you suggest what is the best way to procced? Can I work on this branch directly?

Comment (ahmedjoubest) [reply legacy/Functions/03_functions_energyrequirements.R:99]: Hi @lydialanzoni ,
I just spoke with Yassine — since it's a small change, you can go ahead and put the changes in the same branch, in one or two commits. Just make sure the commit name and description are clear so we can quickly verify that everything works as intended.

Comment (lydialanzoni) [reply legacy/Functions/03_functions_energyrequirements.R:99]: @elaouniyassine @ahmedjoubest
I introduced the change reported above :)

File: R/energy_requirements_core_model.R [hunk 2]
```diff
@@ -101,38 +101,40 @@ calc_net_energy_maintenance <- function(
 calc_net_energy_activity <- function(
     animal,
     cohort,
-    past_man_frac,
-    mmspasture,
     nemain,
     average_weight,
-    offtake_rate
+    offtake_rate,
```
Comment (elaouniyassine) [inline R/energy_requirements_core_model.R:106]: Hi @lydialanzoni,
When changing the function arguments, we need to edit the function documentation.
Could you verify if the new documentation for the new arg names makes sense, thanks!

**Comment by elaouniyassine** [comment]
We will be using another PR #53, this one will be closed
